### DISCUSSION TITLE
OSX temps

### DISCUSSION THREAD

**Discussion by aristocratos** [aristocratos/btop#116] [General]
@joske 
Looking at the sensors we get, pACC and eACC combined makes up ~~8~~ (edit 9 :\ ) values on my mac mini m1. I thought it was supposed to have 4 performance cores and 4 efficiency cores but who knows... But they might be the best fit for mapping to cores, if we can figure out which core id's is for performance and efficiency.

Another way might be to skip displaying coretemps and calculate the average over all tdie values (why are there only 7?) or pACC/eACC and show as main cpu temp.

Thoughts?

**Comment by joske** [comment]
Well, I have no idea. I've looked at many other projects, and most of them just throw their arms in the air and provide a way to show all sensors. You can install the Hot.app and choose view all sensors to quickly assess what is available.

The tdie ones have a few missing, the tdev are 8, but some give negative numbers. Then there's the eACC and pACC, but they are weirdly numbered (and don't make sense since there are supposed to be 4 efficiency cores). It's a complete mess.

So you can choose what you like best, I have no preference, none of them are perfect (or I would have done so already ;)). Until these are documented (which will be never) we will not know what they all mean, it's just guessing).

**Comment by aristocratos** [comment]
Yeah, I'm gonna disable coretemp for M1's and calculate the main CPU temp from an average of all pACC and eACC, seems most sensible.

**Comment by joske** [comment]
updated the M1 MBA to monterey in the hope new firmware/OS would fix the sensors, but get the same things...

**Comment by joske** [comment]
now with macOS 12, gcc no longer finds CoreFoundation etc, so can no longer compile on the M1 :-/

Edit: had to reinstall xcode command line tools, and fine again.

**Comment by aristocratos** [comment]
Huh, well thanks for the warning. Just googling it looks like this is not the first time there's issues with gcc + missing libraries after a macos update.
Xcode updated also?

If you know the paths for the missing libraries I could pack them from my system and you could try manually adding them back.

**Comment by joske** [comment]
No no update top Xcode. Just the command line tools.
